.github/workflows/update-infrastructure-repo.yaml

```
@@ -122,3 +127,40 @@ jobs:
           github-token: ${{ secrets.approve-pr-token }}
           number: ${{ steps.cpr.outputs.pull-request-number }}
           repo: ${{ inputs.source-github-repo }}
+
+      - name: Close existing Pull-requests
+        if: ${{ inputs.close-existing-prs }}
+        env:
+          GH_TOKEN: ${{ steps.generate_token.outputs.token }}
+        run: |
+          # Saner programming environment. Turns common developer mistakes into errors.
+          set -eu -o pipefail
+
+          # Enable debug-logging, prints every command before running it
+          set -x
+
+          if [[ "${{ inputs.service }}" == "" || "${{ inputs.service }}" == "service" ]]; then
+              echo "No explicit service set, not closing old existing pull-requests"
+              exit 0
+          fi
+
+          cleanup() {
+            rm -f open-prs.json || :
+            rm -f old-prs.json || :
+          }
+          trap cleanup EXIT
+
+
+          # List all open PRs for this service/environment-combination
+          gh pr list --label "service:${{ inputs.service }}" --label "environment:${{ matrix.environment }}" --label deployment --state=open --app 'writebackinpr' --repo "${{ github.server_url }}/${{ inputs.source-github-repo }}" --json number > open-prs.json
+          cat open-prs.json
```

```suggestion
          cat open-prs.json | jq
```
This will cause a the process to stop if it is invalid json, which is want we want i guess

---

Added, thanks :)